Paper: Scientific Publishing with MyST Markdown

rowanc1 opened the pull request:
If you are creating this PR in order to submit a draft of your paper, please name your PR with `Paper: <title>`. An editor will then add a `paper` label and GitHub Actions will be run to check and build your paper.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#preview-your-paper) for more information.

_Editor_: <!--editor-->Hongsup Shin @hongsupshin  <!--end-editor-->

_Reviewers_: <!--reviewers-list--> 

- Andy Terrel @aterrel
- Pranoy Ray @pranoy-ray 

<!--end-reviewers-list-->

rowanc1 committed:
🚀 Initial draft of paper for SciPy review

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+In the face of mounting global challenges such as climate change, pandemics, and water security, the imperative for rapid, effective scientific discovery and dissemination has never been more acute. The pace at which these problems evolve and impact societies worldwide demands an equally dynamic and innovative approach to how scientific research is published. Despite significant advancements in t ...
+
+This paper documents some of the design decisions made to address these challenges within two integrated tools: (1) MyST Markdown (Markedly Structured Text, <https://mystmd.org>), a community-run open-source project[^myst-jep], which is a text-based authoring framework that integrates computational content (e.g. Jupyter Notebooks); and (2) Curvenote (<https://curvenote.com>), which is a set of op ...
```
```suggestion
This paper documents some of the design decisions made to address these challenges within two integrated tools: (1) MyST Markdown (Markedly Structured Text, <https://mystmd.org>), a community-run open-source project[^myst-jep], which is a text-based authoring framework that integrates computational content (e.g. Jupyter Notebooks); and (2) Curvenote (<https://curvenote.com>), which is a set of open-source utilities, command-line tools, actions and services[^curvenote] aimed to improve scientific publishing by journals, societies, lab-groups, and individuals. In this article we provide background, motivation and perspective for our efforts in developing new open-source tools for science communication, with examples ranging from individual authors to journal administrators. We give an overview of MyST Markdown but would like to reiterate that MyST Markdown is a community-run project, we do not speak for all of the participants, and the community has varied goals for the project (including API documentation, community guidelines, educational tutorials). Our focus in this article is to give our perspectives on scientific writing and publishing and how it intersects with these open-community projects in addition to the open-source efforts that Curvenote is undertaking around scientific publishing.
```

This addresses a typo: "...tools for science communication as well as _provide examples individuals_, lab-groups, proceedings, and society journals" while also eliminating some duplication from the previous sentence.

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+### Authoring Structured Content
+
+There are currently many challenges for individuals or groups to author research information that can be shared in a structured and rigorous way. By this we mean the things that _structurally_ set a scientific article apart from, for example, a blog post: structured content, cross-references, valid citations with persistent identifiers (PIDs), and standardized metadata for licensing, funding info ...
```
Here, we are explaining how "structure" is a critical component of published research, and part of that is "structured content" - this is a little circular. What actually is "structured content?" Well-defined sections like Introduction, Conclusion, etc? Or machine readability (e.g. XML, JSON)?

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+There are currently many challenges for individuals or groups to author research information that can be shared in a structured and rigorous way. By this we mean the things that _structurally_ set a scientific article apart from, for example, a blog post: structured content, cross-references, valid citations with persistent identifiers (PIDs), and standardized metadata for licensing, funding info ...
+One metric for measuring the difficulty of satisfying these scientific standards is to look at the direct costs that are spent on transforming author submissions (e.g. a PDF or a Word Document) into something that conforms to these standards and is ultimately archived.
+In scientific publishing, about 15% of Article Processing Charges (APCs) go to direct publication costs[^pub-costs] [@10.12688/f1000research.27468.2], which provides a _very_ rough global estimate of over USD\$2 billion dollars[^pub-revenue] that is spent on transforming author submissions (e.g. a word-document, LaTeX, or a PDF) into a copyedited, well-formatted, typeset document that can be arch ...
```
```suggestion
In scientific publishing, about 15% of Article Processing Charges (APCs) go to direct publication costs[^pub-costs] [@10.12688/f1000research.27468.2], which provides a _very_ rough global estimate of over USD\$2 billion dollars[^pub-revenue] that is spent on transforming author submissions (e.g. a word-document, LaTeX, or a PDF) into a copyedited, well-formatted, typeset document that can be archived with appropriate metadata [@10.4045/tidsskr.20.0118]; this estimate does not include the approximately USD\$230 million spent on reformatting articles by scientists _before_ publication [@10.1186/s12916-023-02882-y]. Many of these processes are hidden from authors[^hidden-processes] as well as actionable access to many of the benefits of structured data beyond citation graphs (e.g. "How many citations did this paper get?").
```

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+### Computational Articles
+
+A compounding challenge to scientific publishing that we are exploring through MyST Markdown and Curvenote is how to deeply integrate computational workflows and content into science communication to promote interactive explorations and reproducible practices. There are a host of challenges to from user-interface design, to maintenance, to archiving computational content. Many other tools have wo ...
```
```suggestion
A compounding challenge to scientific publishing that we are exploring through MyST Markdown and Curvenote is how to deeply integrate computational workflows and content into science communication to promote interactive explorations and reproducible practices. There are a host of challenges from user-interface design, to maintenance, to archiving computational content. Many other tools have worked on aspects of integrating computation into scientific articles, notably R-Markdown [@10.1201/9781138359444] and it's successor Quarto (<https://quarto.org>); both of these projects have similar aims to MyST Markdown.
```

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+These questions require authoring tools to be able to execute content, to integrate and display computational/interactive outputs directly in reading experiences, as well as scientific publishing systems that can understand and archive computational content (e.g. Docker containers).
+This deep integration can open up possibilities of embedding interactive visualizations and computational notebooks directly into scientific documents [e.g. @eg:interactivity], transforming articles from static texts into rich, interactive, reproducible narratives.
+In 2023, the authors helped to lead several working groups related to these challenges _Notebooks Now!_, a Sloan Foundation funded project led by the American Geophysical Union.
```
```suggestion
In 2023, the authors helped to lead several working groups related to these challenges as part of _Notebooks Now!_, a Sloan Foundation funded project led by the American Geophysical Union.
```

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+These questions require authoring tools to be able to execute content, to integrate and display computational/interactive outputs directly in reading experiences, as well as scientific publishing systems that can understand and archive computational content (e.g. Docker containers).
+This deep integration can open up possibilities of embedding interactive visualizations and computational notebooks directly into scientific documents [e.g. @eg:interactivity], transforming articles from static texts into rich, interactive, reproducible narratives.
+In 2023, the authors helped to lead several working groups related to these challenges _Notebooks Now!_, a Sloan Foundation funded project led by the American Geophysical Union.
+Those working groups found that integrating computational documents, via Jupyter Notebooks, into scholarly publishing system requires a re-imagination of the publishing processes (from submission to peer-review to production to reading) and that many existing processes, platforms are ill-equipped to handle computational articles [c.f. @10.1029/2023EA003458].
```
```suggestion
Those working groups found that integrating computational documents, via Jupyter Notebooks, into scholarly publishing system requires a re-imagination of the publishing processes (from submission to peer-review to production to reading) and that many existing processes and platforms are ill-equipped to handle computational articles [c.f. @10.1029/2023EA003458].
```

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+## Authoring Tools
+
+MyST Markdown (Markedly Structured Text, <https://mystmd.org>) is a community-driven markup language that is a superset of [CommonMark](https://commonmark.org/) (a standard form of Markdown) with special syntax for citations, cross-references, and block and inline extension points called "directives" and "roles". The block-level content provides multi-line containers surrounded by either backtick ...
```
```suggestion
MyST Markdown (Markedly Structured Text, <https://mystmd.org>) is a community-driven markup language that is a superset of [CommonMark](https://commonmark.org/) (a standard form of Markdown) with special syntax for citations, cross-references, and block and inline extension points called "directives" and "roles". The block-level content provides multi-line containers surrounded by either backticks or colons; examples include callout panels, figures, equations and tables (see [documentation](https://mystmd.org)). There is also specialized support for the types of metadata that are important to collect for scientific articles (funding, ORCIDs, CRediT Roles, etc.).
In 2022, the Executable Books project (<https://executablebooks.org>, which hosts Jupyter Book and MyST) started work on the `mystmd` command line interface (CLI), which was initially developed as the [Curvenote CLI](https://github.com/curvenote/curvenote), and later transferred to the ExecutableBooks project. This tool allows authors writing in MyST Markdown to easily build websites and documents and supports the JupyterLab MyST plugin.
```

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+There are also many similarities of MyST Markdown to R-Markdown [@10.1201/9781138359444], Pandoc, and Quarto (<https://quarto.org>), especially in citation syntax and frontmatter structure.
+The initial use case driving the development and design of MyST Markdown has been [JupyterBook](https://jupyterbook.org), which can create educational online textbooks and tutorials with Jupyter Notebooks and narrative content written in MyST.
+
+In 2022, the Executable Books project (<https://executablebooks.org>, which hosts Jupyter Book and MyST) started work on `mystmd` command line interface (CLI), which was initially developed as the [Curvenote CLI](https://github.com/curvenote/curvenote), and later transferred to the ExecutableBooks project. At its core, MyST adheres to the [CommonMark](https://commonmark.org/) standard, ensuring a ...
```
```suggestion
```

This paragraph has a lot of repetition from the previous paragraphs in this section. With a few small, suggested additions to the previous paragraphs, I think we can just remove this one.

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+## Publishing
+
+There are many ways to publish a MyST Markdown site. The native way that MyST is represented is through content-as-data, which is a collection of JSON files that represent the full documents and all associated metadata (if you are reading this online, you can add a `.json` to the URL to access the content), and enables dynamic serving of the content in a number of ways and when run locally, the M ...
```
```suggestion
There are many ways to publish a MyST Markdown site. The native way that MyST is represented is through content-as-data, a collection of JSON files that represent the full documents and all associated metadata (if you are reading this online, you can add a `.json` to the URL to access the content). This MyST content may be served alongside an independent website theme, which dynamically creates HTML based on these JSON files (very similar to XML-based single source publishing [@10.1109/MITP.2003.1176491], but using modern web-tooling). This server-side approach has a number of advantages, in that it allows you to maintain a journal/site theme, without having to upgrade or rebuild all content as would be required by static tooling. It also puts content addressability as a first class concern (via the `.json`), enabling global cross referencing and opening up opportunities for varied publishing models in future. This is the approach that we take with Curvenote journals, and provide managed services to maintain journal sites, manage and curate content, as well as provide the editorial management tools and workflows.
```

The second sentence of this paragraph was very long and confusing. Is anything important lost in my suggested edit? (For example, I removed "when run locally" - does that distinction matter?)

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+The open-source Curvenote CLI (https://github.com/curvenote/curvenote) provides checks for the structure of a document to ensure it meets automated quality controls for things like references, valid links, author identifiers (e.g. ORCID), or funding information. Executing `curvenote check` in a MyST project will build the project and use the structured data to assess metadata (e.g. do authors pro ...
+
+We have designed these checks in a similar pattern to linting and/or unit-tests in scientific software, which continually give feedback on the structural-health of a code-base with near immediate feedback to authors [@eg:checks]. Authors can take action to improve their metadata directly, including DOIs, CRediT Roles, ORCIDs and structural checks such as word-count or missing sections. For exampl ...
```
```suggestion
We have designed these checks in a similar pattern to linting and/or unit tests in scientific software, which continually give feedback on the structural health of a codebase with near immediate feedback to authors [@eg:checks]. Authors can take action to improve their metadata directly, including DOIs, CRediT Roles, ORCIDs and structural checks such as word count or missing sections. For example, in the SciPy Proceedings in 2024, which used Curvenote checks for their submission system, required and optional metadata was improved by authors reacting to these automated checks without any intervention from the proceedings editorial team (e.g. [](https://github.com/scipy-conference/scipy_proceedings/pull/915) added CRediT roles, ORCIDs abbreviations, and DOIs to get a passing check [See @fig:actions]). This is a low-friction way of improving metadata at the time of authoring or submission to elevate content to the standards that are required for a certain type of publication (e.g. proceedings vs blog post vs peer-reviewed journal).
```

This removes a few excessive hyphens and fixes the typo "...content to ~standards~ the standards that are..."

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+### Enabling Researchers
+
+Although SciPy submissions are made in a central repository, the Curvenote checks can be setup on any repository.
+As MyST Markdown bring a researcher's writing back into a simple text based format, and if they are working with Jupyter notebooks, integrates directly with their computational analysis and results, a researcher's manuscript can be easily created and maintained as an integral part of their research code base and updated in lock step with it. Imagine writing your next paper for submission to SciPy ...
```
```suggestion
As MyST Markdown brings a researcher's writing back into a simple text based format, and if they are working with Jupyter notebooks, integrates directly with their computational analysis and results, a researcher's manuscript can be easily created and maintained as an integral part of their research code base and updated in lock step with it. Imagine writing your next paper for submission to SciPy in a way where your draft was automatically rebuilt with the latest figures and data tables on every push to `main`, and any issues that you would hit on submission are flagged as they happen and so are more easily fixed, in context. This is just one of the ways that continuous science practices we spoke about in [@sec:continuous-science] can lead to radically improved efficiencies at scale.
```

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+Although SciPy submissions are made in a central repository, the Curvenote checks can be setup on any repository.
+As MyST Markdown bring a researcher's writing back into a simple text based format, and if they are working with Jupyter notebooks, integrates directly with their computational analysis and results, a researcher's manuscript can be easily created and maintained as an integral part of their research code base and updated in lock step with it. Imagine writing your next paper for submission to SciPy ...
+
+Curvenote's tools builds on this theme of enabling researchers to do more and to have more visibility into how their work will appear when submitted and published. The previews generated during the journal submission process, or by the actions on a researchers own repo can also be generated by an author at any time. By running `curvenote submit <journal> --draft` researchers can get a set of chec ...
```
```suggestion
Curvenote's tools build on this theme of enabling researchers to do more and to have more visibility into how their work will appear when submitted and published. The previews generated during the journal submission process, or by the actions on a researchers own repo can also be generated by an author at any time. By running `curvenote submit <journal> --draft` researchers can get a set of check results and a preview in the style of that venue, and where submissions of Computational Articles are permitted, authors and reviewers will even get feedback on reproducible environment check that any interactive figures and notebooks execute as expected.
```

fwkoch commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,259 @@
+
+In this article we have presented some of the goals behind features of MyST Markdown and Curvenote, to support new open-science infrastructure including authoring tools as well as publishing workflows that support checks and automation. To support every next-generation research tool on top of open access knowledge, we need access to **high-quality, structured content**, data and software — not ju ...
+
+There is, of course, an _enormity_ of work ahead of these tools to transform science publishing at scale. We are grateful to our society partners who are changing community practices around publishing to support HTML-first publishing, experimenting with computational articles, and implementing new peer-review workflows. Tools on their own do not make change, but can help to enable it. Improvement ...
```
```suggestion
There is, of course, an _enormity_ of work ahead of these tools to transform science publishing at scale. We are grateful to our society partners who are changing community practices around publishing to support HTML-first publishing, experimenting with computational articles, and implementing new peer-review workflows. Tools on their own do not make change, but can help to enable it. Improvements to scientific publishing require many diverse community efforts to improve the quality and speed of how we communicate knowledge, and ultimately to accelerate scientific progress.
```

fwkoch reviewed (commented):
This is great - very cool to write and submit with the same tools we are writing about. Really shows that something is working!

Most of my comments are minor typos, etc - there were just a few paragraphs I suggested more substantial changes.

Also, while previewing this on my phone, it was problematic that footnotes were not accessible at all... I think it's worth taking sometime on https://github.com/executablebooks/mystmd/issues/926 or anther way to enable footnote pop-ups on mobile. But that's a comment for MyST, not this submission 😄

rowanc1 committed:
📖 Wording updates and clarifications

Co-authored-by: Franklin Koch <franklin@curvenote.com>

rowanc1 committed:
🧹 Improve for review (one sentence per line)

rowanc1 committed:
🗑 Remove unused figure

rowanc1 committed:
🧹 Add roles, remove unused citations

ameyxd commented:
@rowanc1 the check fails for a DOI link with a 403 error, but the link seems correct. Are the DOIs added to the bib file correctly?

rowanc1 commented:
@ameyxd the DOI service seems to be intermittent at the moment. These are passing locally!

I have asked @fwkoch to work on a retry for the DOI links.

lheagy commented:
Thanks for sharing this, I have added my review, and I am excited about the open-review of SciPy Proceedings! It is a solid article and reads well. I made some annotations on a pdf (the irony! 🙃 ), some grammatical things, and some things to think about for the future. 

[Transformative Approaches in Scientific Publishing - SciPy Proceedings.pdf](https://github.com/user-attachments/files/15741645/Transformative.Approaches.in.Scientific.Publishing.-.SciPy.Proceedings.2.pdf)

The idea of "continuous science" is really interesting. I think fleshing this out further would be helpful. As a reader, it is quite easy to jump to the conclusion that some of these ideas about "continuous science" will put a lot of work / expectations on reviewers. Which I don't think is the point? To me, it seems that the argument is more that there are practices / ideas from continuous integration that could help improve workflows in science. But this is a bit nuanced and might be nice to try to make more explicit. 

Congrats on bringing this together! I look forward to seeing it published in the proceedings

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+---
+title: Transformative Approaches in Scientific Publishing
+abstract: |
+  Difficult problems that are globally relevant and urgent - like climate change, pandemics, and water security - require us to radically rethink how we publish and work on scientific breakthroughs.
```
```suggestion
  Difficult problems that are globally relevant and urgent — like climate change, pandemics, and water security — require us to radically rethink how we publish and work towards scientific breakthroughs.
```

rowanc1 replied:
added.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+abstract: |
+  Difficult problems that are globally relevant and urgent - like climate change, pandemics, and water security - require us to radically rethink how we publish and work on scientific breakthroughs.
+  Advances in technology for data workflows have increased the speed and scope of scientific discovery.
+  However, scientific dialogue still uses outdated technology for communicating and sharing knowledge, relying on static PDFs that are a poor representation of the complexities of science.
```
```suggestion
  However, scientific dialogue still uses outdated technology for communicating and sharing knowledge, relying on static PDFs that poorly represent the complexities of science.
```

rowanc1 replied:
Applied - thank you.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+  Difficult problems that are globally relevant and urgent - like climate change, pandemics, and water security - require us to radically rethink how we publish and work on scientific breakthroughs.
+  Advances in technology for data workflows have increased the speed and scope of scientific discovery.
+  However, scientific dialogue still uses outdated technology for communicating and sharing knowledge, relying on static PDFs that are a poor representation of the complexities of science.
+  This gap impedes the speed of research dissemination, reuse, and uptake: we require new mediums to compose ideas and ways to share research findings as early and as often as possible; this gap is even more evident in the computational sciences.
```
```suggestion
  This gap impedes the speed of research dissemination, reuse, and uptake; we require new mediums to compose ideas and ways to share research findings as early and as often as possible. This gap is even _more_ evident in the computational sciences.
```

Emphasis OK?

rowanc1 replied:
Going to leave as is.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+In the face of mounting global challenges such as climate change, pandemics, and water security, the imperative for rapid, effective scientific discovery and dissemination has never been more acute.
+The pace at which these problems evolve and impact societies worldwide demands an equally dynamic and innovative approach to how scientific research is published.
+Despite significant advancements in technology that enhance data collection, analysis, and workflow efficiency, the mechanisms through which scientific knowledge is shared have remained largely unchanged for decades [@10.4230/DagMan.1.1.41].
```
```suggestion
Despite significant advancements in technologies that enhance data collection, analysis, and workflow efficiency, the mechanisms through which scientific knowledge is shared have remained largely unchanged for decades [@10.4230/DagMan.1.1.41].
```

rowanc1 replied:
Fixed - thank you.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+In the face of mounting global challenges such as climate change, pandemics, and water security, the imperative for rapid, effective scientific discovery and dissemination has never been more acute.
+The pace at which these problems evolve and impact societies worldwide demands an equally dynamic and innovative approach to how scientific research is published.
+Despite significant advancements in technology that enhance data collection, analysis, and workflow efficiency, the mechanisms through which scientific knowledge is shared have remained largely unchanged for decades [@10.4230/DagMan.1.1.41].
+The prevalent reliance on static PDF formats for research papers starkly contrasts with the complex, data-driven and increasingly computational nature of modern science, creating bottlenecks in knowledge dissemination and uptake.
```
```suggestion
The widespread reliance on static PDF formats for research papers starkly contrasts with the complex, data-driven and increasingly computational nature of modern science, creating bottlenecks in knowledge dissemination and uptake.
```

rowanc1 replied:
Fixed - thank you.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+This paper documents some of the design decisions made to address these challenges within two integrated tools: (1) MyST Markdown (Markedly Structured Text, <https://mystmd.org>), a community-run open-source project[^myst-jep], which is a text-based authoring framework that integrates computational content (e.g. Jupyter Notebooks); and (2) Curvenote (<https://curvenote.com>), which is a set of op ...
+In this article we provide background, motivation and perspective for our efforts in developing new open-source tools for science communication, with examples ranging from individual authors to journal administrators.
+We give an overview of MyST Markdown but would like to reiterate that MyST Markdown is a community-run project, we do not speak for all of the participants, and the community has varied goals for the project (including API documentation, community guidelines, educational tutorials).
```
```suggestion
Though we present an overview of MyST Markdown, it should be emphasized that MyST Markdown is a community-run project and Curvenote do not speak for all of the stakeholders; the community has varied goals for the project (including API documentation, community guidelines, educational tutorials).
```

rowanc1 replied:
Fixed to something closer to this.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+This paper documents some of the design decisions made to address these challenges within two integrated tools: (1) MyST Markdown (Markedly Structured Text, <https://mystmd.org>), a community-run open-source project[^myst-jep], which is a text-based authoring framework that integrates computational content (e.g. Jupyter Notebooks); and (2) Curvenote (<https://curvenote.com>), which is a set of op ...
+In this article we provide background, motivation and perspective for our efforts in developing new open-source tools for science communication, with examples ranging from individual authors to journal administrators.
+We give an overview of MyST Markdown but would like to reiterate that MyST Markdown is a community-run project, we do not speak for all of the participants, and the community has varied goals for the project (including API documentation, community guidelines, educational tutorials).
+Our focus in this article is to give our perspectives on scientific writing and publishing and how it intersects with these open-community projects in addition to the open-source efforts that Curvenote is undertaking around scientific publishing.
```
I think this could be reworded to be clearer, but I'm not confident in making a suggestion!

hongsupshin replied:
I agree with @agoose77 . I think we can be more specific on why.

rowanc1 replied:
I have updated the text here, but am not totally clear on what you are asking for!? Can you clarify?

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+[^myst-jep]: The MyST Markdown project is currently hosted by Executable Books (<https://executablebooks.org>), and we are in the process of moving this into the Jupyter Community through a Jupyter Enhancement Proposal [](https://github.com/jupyter/enhancement-proposals/pull/123).
+[^curvenote]:
+    Curvenote is a company that provides many different tools for authoring and publishing content, including a collaborative WYSIWYG online editor that can export to MyST Markdown.
+    In this article we only discuss our open-source tools as well as highlight relevant experiences from Curvenote's partners when they relate to general ideas for improving scientific publishing.
```
```suggestion
    In this article, we discuss solely our open-source tools and highlight ideas from working with Curvenote's partners that pertain to improving scientific publishing.
```

rowanc1 replied:
Fixed.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+    Curvenote is a company that provides many different tools for authoring and publishing content, including a collaborative WYSIWYG online editor that can export to MyST Markdown.
+    In this article we only discuss our open-source tools as well as highlight relevant experiences from Curvenote's partners when they relate to general ideas for improving scientific publishing.
+
+Our goals in developing these integrated tools and workflows address the related challenges of _authoring_ and _publishing_ in the context of computational, open-science documents and lowering the barrier to continuously releasing and iterating on scientific ideas in the open.
```
```suggestion
In developing these integrated tools and workflows, it is our goal to address the related challenges of _authoring_ and _publishing_ in the context of computational, open-science documents· Through this work, we aim to lowering the barrier to continuously releasing and iterating on scientific ideas in the open.
```

rowanc1 replied:
Fixed.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+    In this article we only discuss our open-source tools as well as highlight relevant experiences from Curvenote's partners when they relate to general ideas for improving scientific publishing.
+
+Our goals in developing these integrated tools and workflows address the related challenges of _authoring_ and _publishing_ in the context of computational, open-science documents and lowering the barrier to continuously releasing and iterating on scientific ideas in the open.
+Introducing authoring tools that allow for structured, interactive and computational content as researchers are writing can additionally change the way it is checked, shared and published — enabling faster iterations and direct ties to reproducible, interactive content.
```
```suggestion
Introducing authoring tools that can understand and express structured, interactive and computational content has the potential to fundamentally change the way scientific writing is checked, shared and published — enabling faster iterations and direct ties to reproducible, interactive content.
```

agoose77 replied:
This drops the "as researchers are writing", which might be a point you wanted to emphasise.

rowanc1 replied:
I think it is redundant here - taking your suggestion. thanks!

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+### Authoring Structured Content
+
+There are currently many challenges for individuals or groups to author research information that can be shared in a structured and rigorous way.
```
```suggestion
There are currently many challenges associated with authoring research information that can be shared in a structured and rigorous way.
```

rowanc1 replied:
leaving as is.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+There are currently many challenges for individuals or groups to author research information that can be shared in a structured and rigorous way.
+By this we mean the things that _structurally_ set a scientific article apart from, for example, a blog post: structured content, cross-references, valid citations with persistent identifiers (PIDs), and standardized metadata for licensing, funding information, authors and affiliations.
+This structured content and metadata as well as the standards behind them are what defines the "scientific record" and enables archiving, discoverability, accessibility, interoperability and the ability to reuse or cite content [@10.1038/sdata.2016.18].
```
```suggestion
These structured content and metadata, as well as the standards behind them, are what define the "scientific record" and enable archiving, discoverability, accessibility, interoperability and the ability to reuse or cite content [@10.1038/sdata.2016.18].
```

rowanc1 replied:
fixed.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+By this we mean the things that _structurally_ set a scientific article apart from, for example, a blog post: structured content, cross-references, valid citations with persistent identifiers (PIDs), and standardized metadata for licensing, funding information, authors and affiliations.
+This structured content and metadata as well as the standards behind them are what defines the "scientific record" and enables archiving, discoverability, accessibility, interoperability and the ability to reuse or cite content [@10.1038/sdata.2016.18].
+One metric for measuring the difficulty of satisfying these scientific standards is to look at the direct costs that are spent on transforming author submissions (e.g. a PDF or a Word Document) into something that conforms to these standards and is ultimately archived.
+In scientific publishing, about 15% of Article Processing Charges (APCs) go to direct publication costs[^pub-costs] [@10.12688/f1000research.27468.2], which provides a _very_ rough global estimate of over USD\$2 billion dollars[^pub-revenue] that is spent on transforming author submissions (e.g. a word-document, LaTeX, or a PDF) into a copyedited, well-formatted, typeset document that can be arch ...
```
```suggestion
In scientific publishing, about 15% of Article Processing Charges (APCs) go to direct publication costs[^pub-costs] [@10.12688/f1000research.27468.2]. When applied to the global publishing industry, this figure suggests that approximately USD\$2 billion dollars[^pub-revenue] is spent on transforming author submissions (e.g. a word-document, LaTeX, or a PDF) into a copy-edited, well-formatted, typeset document that can be archived with appropriate metadata [@10.4045/tidsskr.20.0118]; this estimate does not include the approximately USD\$230 million spent on reformatting articles by scientists _before_ publication [@10.1186/s12916-023-02882-y].
```

I did remove the "very rough" emphasis... we might want that (or use a footnote?).

rowanc1 replied:
I like the rewords, updated locally.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+This structured content and metadata as well as the standards behind them are what defines the "scientific record" and enables archiving, discoverability, accessibility, interoperability and the ability to reuse or cite content [@10.1038/sdata.2016.18].
+One metric for measuring the difficulty of satisfying these scientific standards is to look at the direct costs that are spent on transforming author submissions (e.g. a PDF or a Word Document) into something that conforms to these standards and is ultimately archived.
+In scientific publishing, about 15% of Article Processing Charges (APCs) go to direct publication costs[^pub-costs] [@10.12688/f1000research.27468.2], which provides a _very_ rough global estimate of over USD\$2 billion dollars[^pub-revenue] that is spent on transforming author submissions (e.g. a word-document, LaTeX, or a PDF) into a copyedited, well-formatted, typeset document that can be arch ...
+Many of these processes are hidden from authors[^hidden-processes] as well as actionable access to many of the benefits of structured data beyond citation graphs (e.g. "How many citations did this paper get?").
```
What does 
> actionable access

mean here?

My inference is that we're saying both the processes _and_ benefits of structured data authorship are lost. If so, let's reword.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+    Much of the production publication processes are hidden from scientific authors, with typesetting focused on cross-references, linking citations, ensuring citations have appropriate IDs (e.g. DOIs) as well as conversion to JATS XML (a NISO standard for archiving scientific articles), metadata preparation to [CrossRef](https://crossref.org), and archiving services like LOCKSS (<https://lockss. ...
+    Additionally, the many proprietary services and tools to create both online and PDF outputs of the authors work that are nicely typeset for reading on the web or online.
+
+One goal of the MyST Markdown project is to _dramatically_ reduce these direct-publication costs[^zero-cost] and directly provide structured data as an output of authoring as well as the ability to export content in a variety of formats including HTML, PDF and JATS-XML (a NISO standard for archiving scientific articles).
```
```suggestion
One goal of the MyST Markdown project is to _dramatically_ reduce these direct-publication costs[^zero-cost] and provide direct unfettered access to structured data as an output of authoring. It aims to leverage the availability of this structured representation to export content to a wide variety of formats including HTML, PDF and JATS-XML (a NISO standard for archiving scientific articles).
```

Let me know how `unfettered` sits.

rowanc1 replied:
like it!

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+    Additionally, the many proprietary services and tools to create both online and PDF outputs of the authors work that are nicely typeset for reading on the web or online.
+
+One goal of the MyST Markdown project is to _dramatically_ reduce these direct-publication costs[^zero-cost] and directly provide structured data as an output of authoring as well as the ability to export content in a variety of formats including HTML, PDF and JATS-XML (a NISO standard for archiving scientific articles).
+In this article, we will demonstrate that having structured data throughout authoring leads to a number of novel reading and authoring experiences [e.g. @eg:hover], can provide new opportunities for reuse and quality checks when publishing [e.g. @eg:checks].
```
```suggestion
In this article, we will demonstrate that having structured data throughout authoring can lead to a number of novel reading and authoring experiences [e.g. @eg:hover], and can provide new opportunities for reuse and quality checks when publishing [e.g. @eg:checks].
```

rowanc1 replied:
fixed.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+One goal of the MyST Markdown project is to _dramatically_ reduce these direct-publication costs[^zero-cost] and directly provide structured data as an output of authoring as well as the ability to export content in a variety of formats including HTML, PDF and JATS-XML (a NISO standard for archiving scientific articles).
+In this article, we will demonstrate that having structured data throughout authoring leads to a number of novel reading and authoring experiences [e.g. @eg:hover], can provide new opportunities for reuse and quality checks when publishing [e.g. @eg:checks].
+Furthermore, these transformation processes can be run _continuously_ opening the possibilities for faster feedback [See @sec:continuous-science], iterative drafts, small tweaks and versioned improvements that otherwise would not be worth the time and cost.
```
```suggestion
Furthermore, these transformation processes can be run _continuously_, opening the possibilities for faster feedback [See @sec:continuous-science], iterative drafts, small tweaks and versioned improvements that otherwise would not be worth the time and cost.
```

rowanc1 replied:
fixed.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+Many other tools have worked on aspects of integrating computation into scientific articles, notably R-Markdown [@10.1201/9781138359444] and it's successor Quarto (<https://quarto.org>); both of these projects have similar aims to MyST Markdown.
+From a user-experience goals perspective, we are interested in questions such as:
+
+- how to make a change in a notebook figure and have that immediately show up in a document?
+- how to ensure that a p-value, for example, is computed and inserted directly from a computation, rather than through copy-and-paste?
+- how to expose interactivity and exploration that a researcher often has when analyzing a data-set?
+- how to provide and launch archived interactive computing environments?
```
```suggestion
- How can an author make a change in a notebook figure and have that immediately show up in a document?
- How can a researcher ensure that a p-value, for example, is computed and inserted directly from a computation, rather than through copy-and-paste?
- How can a reader access the interactivity and exploration that a researcher leveraged during the analysis of a dataset?
- How can the scientific community archive interactive documents, and expose them to future readers?
```

This is a big change, please do check you're happy with the language. My main focus was making this a user-story.

rowanc1 replied:
I have changed the wording of these.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+## Authoring Tools
+
+MyST Markdown (Markedly Structured Text, <https://mystmd.org>) is a community-driven markup language that is a superset of [CommonMark](https://commonmark.org/) (a standard form of Markdown) with special syntax for citations, cross-references, and block and inline extension points called "directives" and "roles".
+The block-level content provides multi-line containers surrounded by either backticks or colons; examples include callout panels, figures, equations and tables (see [documentation](https://mystmd.org)).
```
Maybe xref to our mystmd guide here?

rowanc1 replied:
added.

agoose77 commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+This enhanced-links concept can be extended to [Wikipedia](https://en.wikipedia.org/wiki/Wikipedia), RRIDs, RORs, GitHub issues or code, and other scientific databases to augment writing.
+For example, the link `<rrid:SCR_008394>` becomes <rrid:SCR_008394>, with rich-metadata and citations created.
+Wikipedia links come with previews, for example, `<wiki:gravitational_waves>` becomes <wiki:gravitational_waves>.
+GitHub links to pull-requests also give hover information (for example, this example admonition was implemented in [#87](https://github.com/executablebooks/myst-theme/pull/87)).
```
The 
> this example admonition

is no longer directly below/above this paragraph.

rowanc1 replied:
Fixed.

agoose77 reviewed (changes requested):
I got to 2.3 (and stopped before reading it), but wanted to get this feedback in now rather than blocking.

I always find reviews feel a bit "confrontational" -- change this, reword that. So, suffice to say, this is a fab read so far.

stefanv commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+In this scenario, a static HTML site is built from your content which can be hosted as any other static website, while some of he advantages of dynamic hosting are lost, it is an easy and accessible way for individuals to self-publish.
+
+In 2024, Curvenote was asked to improve our integrations to GitHub to support the SciPy Proceedings and re-imagine a MyST based publishing approach that uses GitHub for open-peer-review, implementing a submission, editorial and peer review process with GitHub issues, PRs and actions as a fabric.
+The process previously used technology shared with the Journal of Open Source Software (JOSS), which popularized this approach [@10.7717/peerj-cs.147].
```
Just a factual correction here: the SciPy proceedings tools were implemented around 2010 (I'd have to go check the exact year), and included the proceedings builder (RST to IEEE PDF) and a preview build server (procbuild). The JOSS Whedon bot may have been added into the mix at some point, but SciPy's open, GitHub-based proceedings review, text-format paper build tooling etc. predates JOSS by a significant margin.

/cc @jarrodmillman who has a better memory than me, and who published the first proceedings

rowanc1 replied:
Thank you @stefanv, I will update that to include some of the history. When I was writing that sentence I was only trying to capture the current state rather than evolution, but it does come across as misrepresenting that history a bit. Not my intention!

stefanv replied:
Of course, never thought so! Just think it's worth mentioning, since it was novel at the time (and, in a sad statement about the publishing industry, still is, to a large extent).

cbcunc replied:
I would add that the JOSS Whedon bot was customized and renamed "scoobies" for SciPy Proceedings in 2021. What scoobies did is largely superseded by how we are now using GitHub projects. Here's a listing of the "scoobies help" output:
```
# List all available commands
@scoobies help

# Show our community Code of Conduct and Guidelines
@scoobies code of conduct

# Add to this issue's reviewers list
@scoobies add @username to reviewers

# Remove from this issue's reviewers list
@scoobies remove @username from reviewers

# Assign a user as the editor of this submission
@scoobies assign @username as editor

# Remove the editor assigned to this submission
@scoobies remove editor

# Add a user to this issue's assignees list
@scoobies add assignee: @username

# Remove a user from this issue's assignees list
@scoobies remove assignee: @username

# Builds paper
@scoobies build paper

# Checks build status
@scoobies build status

# Check the references of the paper for missing DOIs
# Optionally, it can be run on a non-default branch 
@scoobies check references
@scoobies check references from branch custom-branch-name

# Label issue with: paper
@scoobies mark paper

# Label issue with: needs-review. Remove labels: unready, needs-more-review, pending-comment, ready
@scoobies mark needs review

# Label issue with: needs-more-review. Remove labels: unready, needs-review, pending-comment, ready
@scoobies mark needs more review

# Label issue with: pending-comment. Remove labels: unready, needs-review, needs-more-review, ready
@scoobies mark pending comment

# Label issue with: ready. Remove labels: unready, needs-review, needs-more-review, pending-comment
@scoobies mark ready

# Label issue with: unready. Remove labels: needs-review, needs-more-review, pending-comment, ready
@scoobies mark not ready

# Label issue with: does_not_build:server
@scoobies mark server fail

# Remove labels: does_not_build:server
@scoobies mark server success
```

cbcunc replied:
@mepa hosted scoobies on Heroku, I believe. And I hosted procbuild on a VM at RENCI using a letsencrypt/certbot cert. There was overhead/maintenance involved.

stefanv replied:
I don't recall where we first hosted procbuild in 2023, maybe on a Berkeley machine or on new.scipy.org, but I know @mpacer around 2018 hosted it on Heroku as well.

mepa replied:
I'll add more detail to what @stefanv and @cbcunc have already mentioned just for interest. (I'm not implying it needs to be incorporated into the current paper.)

First, here is a nice blog post by @deniederhut about the SciPy partnership with JOSS (which was initiated in 2020): https://blog.neater-hut.com/scipy-is-partnering-with-joss-part-1.html

From the SciPy Proceedings Committee side, Dillon integrated JOSS's review management bot in time for the 2021 proceedings review period, and deployed it on Heroku free tier in both 2021 and 2022. By 2023, Heroku had [removed its free product plans](https://help.heroku.com/RSBRUH58/removal-of-heroku-free-product-plans-faq) so I looked for potential free options but ended up deploying it on Heroku.

deniederhut replied:
> SciPy's open, GitHub-based proceedings review, text-format paper build tooling etc. predates JOSS by a significant margin

Indeed, Arfon has cited SciPy as an influence on the design of JOSS's publication system

tkoyama010 replied:
Oh that great! I thought JOSS came first and SciPy was influenced by it.

pranoy-ray commented:
Hey all, I am Pranoy, a graduate researcher at GeorgiaTech; I will be a reviewer for this paper, and it's nice to meet you all. Can someone send me a direct link to the paper? (I mean the pdf or something in a readable format with all the figures incorporated)?

rowanc1 commented:
The html preview is in the second comment!

https://github.com/scipy-conference/scipy_proceedings/pull/918#issuecomment-2142652751

Looking forward to your comments.

aterrel commented:
Hello! I'll have some notes on the review this week, been on vacation and travel the last few.

rowanc1 commented:
Thanks @aterrel and @pranoy-ray -- very much looking forward to your feedback.

natesjacobs commented:
A few comments on this article below. Loved reading this and very excited for the future of MyST and native computational article formats!

**Major / general comments:**

1. I would consider adopting a slightly more specific title / overarching narrative, that isn't "Transformative Approaches in Scientific Publishing" but slightly more nuanced in that it starts from a more realistic place of solving computationally intensive writing (your core strength where you outcompete others) and how THAT can help solve core problem with all scientific publishing: "How code-friendly authoring platforms will transform how all science articles are written". It feels a bit more specific and playing to your core strength, while still making the broad sweeping claim.

2. Consider setting up a more specific relationship between MyST and curvenote, rather than just discussing them as "two different tools doing similar things". I really like the narrative that curvenote is leading an effort to establish MyST as an open standard for structured articles. So MyST is at the core of a movement, and curvenote is adopting it for their products and helping to expand it out into more mainstream use.

3. In regard to Continuous Science Practices and other more idealistic open science concepts, I would make sure to think deeply about past failures when facing the overwhelming force of prestige and career advancement, and make sure that factors into your writing and framing of issues so that the audience doesn't assume naivité.  This is tricky because some audiences like to hear the idealistic pure venting etc (like me!) but I do think a very critical component of success will be answering the question of how a publishing venue or workflow will advance someone's career. Like it or not, that has been the single driving force for all publishing evolutions and lack thereof. It's a hard pill to swallow, since you would hope we would care more about science being correct and rapid, but as far as I can tell the University - Publisher - Funder trifecta still cares deeply about prestige and publishing venue above all else. I can talk more about what I was hoping to build up as an alternative prestige pathway (based on organic growth and earned media which has analogs in other industries like the music industry), but perhaps for this article merely acknowledging it or stating it as an issue would suffice.

4. Consider emphasizing "fun to write and read" or "expressive articles" as a high level take home from the feel of the article. This may not be appropriate for this article, but one gut level concept I always tried to hit (unsuccessfully most times, so take this with a grain of salt!) is that these new types of authoring can just be a lot more fun and expressive. You don't have to collapse your figure into a single static image. You can link to the dynamic content or the code and let people jump right into it! That's a really fun experience for both the author and the reader. Actions that are fun are sticky for users, and products that are sticky tend to be successful. 

**Minor / specific comments:**

1. "Many other tools have worked on aspects of integrating computation into scientific articles, notably R-Markdown Xie et al., 2018 and it’s successor Quarto (https://quarto.org/); both of these projects have similar aims to MyST Markdown." 

This feels VERY markdown heavy. I don't have a comprehensive knowledge of computationally focused article formats, but at a minimum you should also mention eLife's experiment with 'executable articles' as well as the efforts of 'papers with code'. Or, alternatively, you could narrow the scope of this statement to markdown related efforts. 

2.  Double check to make sure it's clear to audience that this is the curvenote team writing about curvenote

aterrel commented:
INITIAL Independent Review Report
===========================

Reviewer: Andy R. Terrel

Department/Center/Division: Compute Products

Institution/University/Company: NVIDIA

Field of interest / expertise: Computer Science / Computational Mathematics

Country: USA


Article reviewed: [Scientific Publishing with MyST Markdown](https://github.com/scipy-conference/scipy_proceedings/pull/918#top)


GENERAL EVALUATION
------------------

Please rate the paper using the following criteria (please use the abbreviation
to the right of the description)::

  below     doesn't meet standards for academic publication
  meets     meets or exceeds the standards for academic publication
  n/a       not applicable

- Quality of the approach: Meets

- Quality of the writing: Meets

- Quality of the figures/tables: Meets


SPECIFIC EVALUATION
-------------------

For the following questions, please respond with 'yes' or 'no'.  If you
answer 'no',  please provide a brief, one- to two-sentence explanation.

- Is the code made publicly available and does the article sufficiently
  describe how to access it?  

PARTIAL, Curvenote.com is a proprietary tool, but the software claims to be open source. In trying to reproduce a local build, the documentation links found in the open source repository https://github.com/curvenote/curvenote, led to broken links on the curvenote.com website, i.e. (https://curvenote.com/docs/cli, https://curvenote.com/docs/web, https://curvenote.com/docs/cli/authorization , ), thus it is hard to verify this claim without fully reviewing the software which is out of scope for this review. 

- Does the article present the problem in an appropriate context?
  Specifically, does it:

  - explain why the problem is important,
  - describe in which situations it arises,
  - outline relevant previous work,
  - provide background information for non-experts

The article does explain the problem but could be better. My recommendation is to modify the following justifications to only scholarly references that are verifiable rather than hand-wavy arguments without sufficient data.

Using climate change to justify a new workflow is hardly scholarly. I do not see justification that faster publications will solve these generational societal problems.  

Additionally the paper claims that it takes "elite" software teams to use standard devops practices, this is motivated by marketing whitepapers trying to sell platforms, not scholarly inquiry. 

Finally the paper relies on an extremely rough economic argument claiming that it can address 15% of the global spend on publishing is not believable. Better publishing tools have been around for decades, but the business models of the publishing world have not changed. To make this argument, the authors would need to give a much more detailed analysis of the economics of these direct costs and evaluate what behavior changes scientists would need to make to achieve this lower cost. If they could use tools that were so much cheaper, it seems they would do so but there are fundamental barriers that are not scholarly studied in this paper.

- Is the content of the paper accessible to a computational scientist
  with no specific knowledge in the given field?

Yes

- Does the paper describe a well-formulated scientific or technical
  achievement?

It gives a concept of continuous scientific publication (although using a more generic and misleading term "continuous science") and how features of the curvenotes cli can support such practices

- Are the technical and scientific decisions well-motivated and
  clearly explained?

The article needs a better discussion on prior methods to reduce costs of publications. While it may be true that the world does spend too much effort on a older process (which the paper used a paper from 14 years ago to justify), it is not true that there are not other low cost publishing alternatives. The argument would be much stronger if it compared itself to these processes and their apparent failure to create the cultural change advocated in scientific publishing that is advocated here.

- Are the code examples (if any) sound, clear, and well-written?

No code examples given.

- Is the paper factually correct?

Yes modulo the arguments above that are made in an unverifiable way.

- Is the language and grammar of sufficient quality?

Yes.

- Are the conclusions justified?

Yes the conclusions are mild in comparison to the initial claims of the paper. 

- Is prior work properly and fully cited?

Need citation on other low-cost publication methodologies.

- Should any part of the article be shortened or expanded? Please explain.

Yes, either justify how faster science solves generational societal problems or remove the claim. The authors need to evaluate other low cost publication methodologies and their gaps rather than just claiming it is a large economic problem and citing a paper from 14 years prior. 

- In your view, is the paper fit for publication in the conference proceedings?
  Please suggest specific improvements and indicate whether you think the
  article needs a significant rewrite (rather than a minor revision).
  
This paper is fit for publication with minor revisions as stated above.

rowanc1 commented:
Thank you very much @natesjacobs and @aterrel for the reviews, appreciate the time that you took and will respond with comments and changes to the manuscript. In terms of timing, I will wait for a further review from @pranoy-ray before I dive into satisfying the changes you have requested and providing more information. Thanks again all for your help!

hongsupshin commented:
Hey @rowanc1 , just got an email from @pranoy-ray saying that he won't be able to conduct the review. I will review the paper and leave the comments this weekend!

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+---
+title: Transformative Approaches in Scientific Publishing
```
As @natesjacobs mentioned, I would also suggest a more specific title to be more accurate and attract more audience.

rowanc1 replied:
I have refined this to focus more on the tools:

`Continuous Tools for Scientific Publishing: Using MyST Markdown and Curvenote to encourage continuous science practices`

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+title: Transformative Approaches in Scientific Publishing
+abstract: |
+  Difficult problems that are globally relevant and urgent - like climate change, pandemics, and water security - require us to radically rethink how we publish and work on scientific breakthroughs.
+  Advances in technology for data workflows have increased the speed and scope of scientific discovery.
```
I think this sentence provides a stronger motivation than the first sentence.

rowanc1 replied:
I have moved this first (locally, will push shortly). Thanks.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+---
+title: Transformative Approaches in Scientific Publishing
+abstract: |
+  Difficult problems that are globally relevant and urgent - like climate change, pandemics, and water security - require us to radically rethink how we publish and work on scientific breakthroughs.
```
You can probably elaborate a bit more on what specific aspects of "Difficult problems" call for rethinking publication. Is this because of the complexity of the problem, large-scale collaboration, higher stakes, etc? The next sentence explains a bit but the link can be strengthened here (if this is the first sentence especially)

rowanc1 replied:
I have revamped the abstract locally to make this clearer!

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+  Advances in technology for data workflows have increased the speed and scope of scientific discovery.
+  However, scientific dialogue still uses outdated technology for communicating and sharing knowledge, relying on static PDFs that are a poor representation of the complexities of science.
+  This gap impedes the speed of research dissemination, reuse, and uptake: we require new mediums to compose ideas and ways to share research findings as early and as often as possible; this gap is even more evident in the computational sciences.
+  In this paper we discuss two tools for scientific dialogue, MyST Markdown and Curvenote, and illustrate examples of improving metadata, the reading experience, and transforming publishing practices for individuals and societies.
```
I think given that the title has the word "transformative" this reasoning can be strengthened. For now, "improving metadata, the reading experience" is why these tools are "transformative", and I am wondering whether the reason is strong enough (providing some details might be helpful?)

I see that the first few paragraphs of the introduction summarizes this well, so maybe bringing some elements from there might be helpful.

rowanc1 replied:
I have brought up elements discussing computation, interactivity and automation through continuous practices.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+By this we mean the things that _structurally_ set a scientific article apart from, for example, a blog post: structured content, cross-references, valid citations with persistent identifiers (PIDs), and standardized metadata for licensing, funding information, authors and affiliations.
+This structured content and metadata as well as the standards behind them are what defines the "scientific record" and enables archiving, discoverability, accessibility, interoperability and the ability to reuse or cite content [@10.1038/sdata.2016.18].
+One metric for measuring the difficulty of satisfying these scientific standards is to look at the direct costs that are spent on transforming author submissions (e.g. a PDF or a Word Document) into something that conforms to these standards and is ultimately archived.
+In scientific publishing, about 15% of Article Processing Charges (APCs) go to direct publication costs[^pub-costs] [@10.12688/f1000research.27468.2], which provides a _very_ rough global estimate of over USD\$2 billion dollars[^pub-revenue] that is spent on transforming author submissions (e.g. a word-document, LaTeX, or a PDF) into a copyedited, well-formatted, typeset document that can be arch ...
```
I do like this analysis!

rowanc1 replied:
🚀 It is crazy, and hopefully something that can be radically changed over the next decade. :)

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+    Additionally, the many proprietary services and tools to create both online and PDF outputs of the authors work that are nicely typeset for reading on the web or online.
+
+One goal of the MyST Markdown project is to _dramatically_ reduce these direct-publication costs[^zero-cost] and directly provide structured data as an output of authoring as well as the ability to export content in a variety of formats including HTML, PDF and JATS-XML (a NISO standard for archiving scientific articles).
+In this article, we will demonstrate that having structured data throughout authoring leads to a number of novel reading and authoring experiences [e.g. @eg:hover], can provide new opportunities for reuse and quality checks when publishing [e.g. @eg:checks].
```
I am wondering how the hover example would look like in static pdf? Also, nitpicky but currently the `[e.g. @eg:checks]` is shown as Example 3. Since the previous example (hover) is Example 1, should this be Example 2 instead of 3?

rowanc1 replied:
I have now introduced them in order, and removed the `e.g.` which is a bit redundant.

![image](https://github.com/user-attachments/assets/aeccb670-a70d-472d-946d-6ee58ad40879)

My plan is to make a different figure for PDF. ... !

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+A compounding challenge to scientific publishing that we are exploring through MyST Markdown and Curvenote is how to deeply integrate computational workflows and content into science communication to promote interactive explorations and reproducible practices.
+There are a host of challenges from user-interface design, to maintenance, to archiving computational content.
+Many other tools have worked on aspects of integrating computation into scientific articles, notably R-Markdown [@10.1201/9781138359444] and it's successor Quarto (<https://quarto.org>); both of these projects have similar aims to MyST Markdown.
```
I am curious whether the paper mentions differences between curvenote/myst and these referred tools

rowanc1 replied:
I have added a few sentences on comparing the projects more specifically.

>MyST is influenced by [reStructuredText (RST)](https://en.wikipedia.org/wiki/reStructuredText) and [Sphinx](https://www.sphinx-doc.org) – pulling on the nomenclature and introducing additional standards where appropriate.
> There are also many intentional syntax similarities of MyST Markdown to R-Markdown [@10.1201/9781138359444], Pandoc, and Quarto (<https://quarto.org>), especially in citation syntax and frontmatter structure.
> MyST Markdown is written in Javascript and builds upon the `unified` community of tools, output formats and transforms, whereas Quarto builds upon Pandoc written in Haskell + Lua; there are also differences in approach in the legal and community foundations (Pandoc and Quarto are GPL vs. MyST which is a more permissive MIT license; MyST is a community-driven project whereas Quarto is driven by a single corporate entity).

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+    In addition to increasing speed and robustness, continuous delivery practices also demonstrated that the elite teams spent 20% more time on new work, had 5-20% less manual work, and were 1.8x more likely to recommend their team as a great place to work ([2018 DORA Report](https://services.google.com/fh/files/misc/state-of-devops-2018.pdf)).
+    These numbers are intriguing when contrasted to researchers, where (a) scientists already work 53.96 hours a week on average, and only about 36% of their time is actually spent on research (8% on grants, 32% on teaching, and 24% on service) [@10.1016/j.econedurev.2007.04.002]; and (b) graduate students are six times more likely to experience depression than the general population [@10.1038/nb ...
+
+The analogy between scientific publishing and software releases is imperfect and non-prescriptive (i.e. scientific research is very different than developing a product).
+However, the analogy is illustrative in areas where there is a focus on iterations, smaller changesets and releasing in-progress work as soon as possible to get feedback from peers (i.e. scientific peer-review) or users (in the case of software products).
```
Glad you mentioned the difference between the two :)

rowanc1 replied:
Thanks - lots of things to learn and think about in the analogy, but certainly not prescriptive.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+The analogy between scientific publishing and software releases is imperfect and non-prescriptive (i.e. scientific research is very different than developing a product).
+However, the analogy is illustrative in areas where there is a focus on iterations, smaller changesets and releasing in-progress work as soon as possible to get feedback from peers (i.e. scientific peer-review) or users (in the case of software products).
+The speed of scientific progress depends _on part_ on the speed of iteration and feedback.
+The time it takes for the peer-review process is over three months, in high profile journals like Nature that time has almost doubled over the past decade [@Powell2016Does].
```
Do you think using curvenote would reduce this? If yes, then perhaps you can elaborate

rowanc1 replied:
I think that changes in publishing practices can reduce this -- i.e. the publish-review-curate model, where you publish a preprint first, archive that, and then support the review and curation.

I have added a link to prereview at the end of this paragraph which is doing more in this space than Curvenote  at the moment.

This would be very interesting to try at SciPy next year if you are game. :)

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+The speed of scientific progress depends _on part_ on the speed of iteration and feedback.
+The time it takes for the peer-review process is over three months, in high profile journals like Nature that time has almost doubled over the past decade [@Powell2016Does].
+Rejections are anywhere from 50-90% [@10.12688/f1000research.27468.2] with valuable reviews and expertise coming months or even years after the work is completed[^time-to-feedback].
+There are wide spread efforts in scientific publishing that focus both on sharing smaller components of research (e.g. FigShare [@10.7771/2380-176X.7269], Octopus [@10.17863/CAM.96819], MicroPublications [@10.1186/2041-1480-5-28], NanoPublications [@10.3233/ISU-2010-0613], Protocols [@10.1371/journal.pbio.1002538], PreRegistrations [@10.1073/pnas.1708274114]) as well as sharing research sooner in ...
```
I am wondering whether OpenReview also fits in this series of efforts.

rowanc1 replied:
I have added a comment about PREreview and a citation to https://www.nature.com/articles/d41586-020-00613-4.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+- When a computational figures or data output changes, how long does it take to integrate that into your document?
+
+In science there is a highly manual, absurdly expensive and disconnected process between authoring and publishing.
+By moving to continuous practices and investing in the appropriate infrastructure to support _continuous science_ we believe there is an opportunity to accelerate scientific discovery by multiple orders of magnitude while simultaneously increasing reproducibility and robustness of the underlying science.
```
In a similar sense, one of the major benefits of this approach is more transparency to the research and review process.

rowanc1 replied:
Changed to:

> increasing reproducibility, robustness and transparency of the underlying science.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+:::{figure} ./images/reuse-jupyter-outputs.png
+:label: fig:reuse
+A schematic of embedding content from Jupyter Notebooks into an article.
+The purple and orange components, interactive figure or other computational outputs, are created in a computational notebook and subsequently used in other narrative or presentation-focused scientific article.
```
What's the difference between the middle and right sketches? Also, blue, green, and red colors are not explained, so it's probably better to explain or simplify by reducing the number of colors (especially red vs. blue because I don't know what the difference should be)

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+### Single Source to Many Outputs
+
+These capabilities of cross-references, typography and embedding visualizations and data-frames are complemented by an single-source export system that supports professional article templates and JATS XML [@fig:export].
+This is referred to as single-source publishing [c.f. @10.1109/MITP.2003.1176491], however, many implementations in scientific publishing focus first on manual translation to XML (e.g. from Word or LaTeX), rather than on an author-facing implementation.
```
Do we need c.f.?

rowanc1 replied:
I have removed these.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+### Single Source to Many Outputs
+
+These capabilities of cross-references, typography and embedding visualizations and data-frames are complemented by an single-source export system that supports professional article templates and JATS XML [@fig:export].
```
an -> a, or drop

rowanc1 replied:
Thanks, fixed!

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+This is the approach that we take with Curvenote journals, and provide managed services to maintain journal sites, manage and curate content, as well as provide the editorial management tools and workflows.
+
+You can also easily self-host a MyST Markdown site using GitHub Pages.
+To create a GitHub pages output you can run `myst init --gh-pages`, which will walk you through the steps of creating a GitHub action to publish your content as a static website.
```
a GitHub pages -> GitHub pages

rowanc1 replied:
Fixed, thanks!

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+You can also easily self-host a MyST Markdown site using GitHub Pages.
+To create a GitHub pages output you can run `myst init --gh-pages`, which will walk you through the steps of creating a GitHub action to publish your content as a static website.
+In this scenario, a static HTML site is built from your content which can be hosted as any other static website, while some of he advantages of dynamic hosting are lost, it is an easy and accessible way for individuals to self-publish.
```
he advantages -> the advantages

rowanc1 replied:
Fixed! Thank you.

hongsupshin commented on papers/cockett_etal/article.md:
````diff
@@ -0,0 +1,372 @@
+The checks in this example have promoted the author to improve metadata, see [](https://github.com/scipy-conference/scipy_proceedings/pull/911).
+```
+
+### Enabling Researchers
````
Perhaps we can rephrase this to be more specific

rowanc1 replied:
I have changed this to "continuous practices" and updated the text around this to talk about the combination of a simplified authoring experience and integrated checks and automation through actions.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+### Enabling Researchers
+
+Although SciPy submissions are made in a central repository, the Curvenote checks can be setup on any repository.
```
This makes the whole section sound like this is specific to SciPy, which I think it's not?

rowanc1 replied:
I have revamped this section, and removed the references to scipy, because as you note, they are much more general.

hongsupshin commented on papers/cockett_etal/article.md:
```diff
@@ -0,0 +1,372 @@
+
+Although SciPy submissions are made in a central repository, the Curvenote checks can be setup on any repository.
+As MyST Markdown brings a researcher's writing back into a simple text based format, and if they are working with Jupyter notebooks, integrates directly with their computational analysis and results, a researcher's manuscript can be easily created and maintained as an integral part of their research code base and updated in lock step with it.
+Imagine writing your next paper for submission to SciPy in a way where your draft was automatically rebuilt with the latest figures and data tables on every push to `main`, and any issues that you would hit on submission are flagged as they happen and so are more easily fixed, in context.
```
I think mentioning SciPy is good but we might want to emphasize that this is not just for SciPy. We can use SciPy as a case study example.

rowanc1 replied:
I have updated this section considerably - thanks for the feedback here!

hongsupshin reviewed (changes requested):
Great article. Suggesting minor changes for improved clarity and typos.

hongsupshin commented:
Hey Rowan, just a friendly reminder that all initial reviews are in. I highly recommend you start responding to the comments soon since it'd take time for the reviewers to respond to the changes. Remember that the open review period ends on Sep 2, and you will not be able to make any changes to the manuscript after that point. If you have any questions, please let me know!

rowanc1 committed:
Address @agoose77 and @lheagy comments

rowanc1 committed:
Updates from reviewers

rowanc1 commented:
Thank you @lheagy for the reviews. I have implemented your spelling and grammar fixes. Additionally, we will improve the figures and the description of continuous science that you suggested.

Thank you @agoose77 for the review and the fixes to the language. I have adopted the majority of your changes and I think they improved the quality of the manuscript.

Thanks @stefanv @cbcunc @mepa @deniederhut for providing some more information on previous versions of the SciPy proceedings infrastructure. I have updated this sentence to better reflect the influence of SciPy Proceedings infrastructure.

---

@aterrel - thanks for the detailed review.

> Is the code made publicly available?

We have updated the text clarifying the scope of this paper and links to the open source repositories:

- https://github.com/jupyter-book/mystmd (community-driven project by Jupyter)
- https://github.com/curvenote/curvenote (CLI - provides checks)
- https://github.com/curvenote/actions (GitHub actions - builds, checks, provides feedback)

We have fixed the links in our Curvenote readme, thanks for catching those.

> Only use scholarly references that are verifiable

We have added several other references when discussing continuous practices. We have also taken a pass through the article to ensure that some of the claims (both economic and thinking about climate change, pandemics etc.) are motivation only.

> Using climate change to justify a new workflow is hardly scholarly.

I have rewritten the abstract and removed this, however, this is relatively common in motivating improvements to scholarly publishing, albeit more on pandemics lately. The argument is that scientists are doing important work, and getting access to that work sooner (e.g. via preprints, open data) in better, more actionable ways (more integrated publishing models, notebooks, integrated compute, etc.) accelerates the time to building upon, understanding or (re)using that work. Some high-profile examples of this motivation are the UN and US policy that are making similar claims on open access to research. For example:

> [Open Science] is increasingly recognized as a critical accelerator for the achievement of the United Nations Sustainable Development Goals.
>
> - United Nations on [Open Science](https://www.unesco.org/en/open-science), 2021

Additionally, the White House making a similarly broad claim in 2022:

> When federally funded research is available to the public, it can improve lives, provide
> policymakers with important evidence with which to make critical decisions, accelerate the rates
> of discovery and translation, and drive more equitable outcomes across every sector of society.
>
> Not only to fight a pandemic, but to advance all areas of study, including urgent issues such as cancer, clean energy, economic disparities, and climate change.
>
> - [Nelson Memo](https://www.whitehouse.gov/wp-content/uploads/2022/08/08-2022-OSTP-Public-access-Memo.pdf) - Office of Science and Technology Policy (White House, 2022)

The Nelson Memo, similarly goes from those high-level statements to mandating requirements on metadata, co-author affiliations, funding sources, and other persistent identifiers and talking about the actual _mechanics_ of open science.

Improved tooling, practices and incentives for open science is critical and can have significant impacts on both the cost and speed of scientific progress.

> I do not see justification that faster publications will solve these generational societal problems.

As an example, faster publication of genomes early in pandemic led to early designs of vaccines within [two days](https://nymag.com/intelligencer/2020/12/moderna-covid-19-vaccine-design.html) and manufacturing within [two weeks](https://covid19.nih.gov/news-and-stories/vaccine-development). The urgency prompted a ["unprecedented"](https://www.nature.com/articles/s41591-021-01654-6) and [lasting](https://www.nature.com/articles/d41586-024-00401-4#ref-CR1) change in how science was shared (via preprints), and major publications temporarily changed their access model to make research [free to read](https://www.springernature.com/gp/researchers/campaigns/coronavirus#c17669228). These changes to enable faster, more open publishing practices ["saves lives"](https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/s12874-021-01304-y); has prompted regulatory change (e.g. the [OSTP Nelson memo](https://www.whitehouse.gov/wp-content/uploads/2022/08/08-2022-OSTP-Public-access-Memo.pdf) which comes into full effect in 5 months); and is continuing to change funding models to focus on [earlier access to research](https://www.nature.com/articles/d41586-024-00996-8) especially via preprints.

We have added a footnote as well as some of these references.

> "elite" software teams & marketing whitepapers

We have rephrased the "elite" software team language and updated the text of this section. The DORA report is the largest industry survey on continuous delivery practices that we know of. We have also added several other references to smaller studies, surveys and review articles showing similar conclusions and speedups.

> claiming that [MyST/Curvenote] can address 15% of the global spend on publishing is not believable

Stranger things have happened! However, we do not make this claim, only that our goal is to reduce these direct-publication costs.

> The article needs a better discussion on prior methods to reduce costs of publications. [...] The argument would be much stronger if it compared itself to these processes and their apparent failure to create the cultural change advocated in scientific publishing that is advocated here.

We have improved a comparison to the Journal of Open Source Software (JOSS).
The economics provide motivation, however, the main take away is that authoring structured content _directly_ provides new, transformed reading experiences and the ability to iterate/release versions close to "publication quality" by ensuring checks, continuous practices, etc.
We have expanded a comparison to Quarto which is the closest tool to doing similar aspects of these workflows, and most especially are in the authors control rather than the publisher.
A wider analysis on the economics and reasons why the sociotechnical publishing system has not embraced lower cost models is beyond the scope of this article.

> used a paper from 14 years ago to justify

This is a manifesto to a conference, I have updated the text with a footnote to make that more clear.

---

Thank you @hongsupshin for the review - I have responded to the majority of your comments inline and resolved them. Feel free to reopen if you have further refinements to make! I am still working through a few more of the issues in the next few days!

Thanks @natesjacobs for the review. I will continue to work through some responses in the next day or two as well.

---

TODO:

- [x] Address remaining comments by @hongsupshin
- [x] Final pass on text

rowanc1 committed:
Ignore DOIs

rowanc1 committed:
Updates addressing @hongsupshin's comments.

hongsupshin reviewed (approved):
Looks good to me. Thanks @rowanc1 for all the hard work!

hongsupshin commented:
@aterrel @natesjacobs @agoose77 Hello reviewers, the author @rowanc1 made changes to the manuscript based on your comments. If you don't have any further comments, could you please approve the changes? Thanks for your time and effort for reviewing this paper!

hongsupshin commented:
Thanks @fwkoch @lheagy @aterrel @stefanv @natesjacobs @agoose77 for reviewing this paper!
